File: SubscriberClient/src/subscriber/common/MessageProcessor.java
package subscriber.common;
 

import java.nio.charset.StandardCharsets;

import com.pcbsys.nirvana.base.nHeader;
import com.pcbsys.nirvana.client.nConsumeEvent;

public class MessageProcessor {

	public void process(nConsumeEvent event) {

		//nHeader header = event.getHeader();
		byte[] eventData = event.getEventData();
		if (eventData != null) {

			String sActualMessage = new String(eventData, StandardCharsets.UTF_8);

			System.out.println("sOutput:" + sActualMessage);
			
			// sActualMessage contains message from Universal messaging topic.
			// Can add relevant code to further process sActualMessage
			

		}
	}
}

File: UniversalMessaging/UniversalMessagingClient/Common/MessageProcessor.cs
using com.pcbsys.nirvana.client;
using System;

namespace UniversalMessagingClient.Common
{
    public class MessageProcessor
    {
        public void Process(nConsumeEvent consumeEvent)
        {
            byte[] buffer = consumeEvent.getEventData();
            if (buffer != null)
            {
                //Add its length to the byte counter

                var output = System.Text.Encoding.UTF8.GetString(buffer, 0, buffer.Length);
                Console.WriteLine("Message Received -- :  " + output);
            }
        }

    }
}
